x-pack/plugins/fleet/public/components/agent_enrollment_flyout/confirm_agent_enrollment.tsx

```diff
@@ -5,39 +5,108 @@
  * 2.0.
  */
 
-import React from 'react';
-import { EuiCallOut, EuiButton } from '@elastic/eui';
+import React, { useEffect, useRef, useState } from 'react';
+import { EuiCallOut, EuiButton, EuiText, EuiLink } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
 import { i18n } from '@kbn/i18n';
 
-import { useGetAgentStatus } from '../../hooks';
+import { sendGetAgents } from '../../hooks';
 import { AGENTS_PREFIX } from '../../constants';
 interface Props {
-  policyId: string;
-  onClickViewAgents: () => void;
+  policyId?: string;
+  troubleshootLink: string;
+  onClickViewAgents?: () => void;
+  agentCount: number;
+  agentEnrolled: boolean;
+  setAgentEnrollment: (v: boolean) => void;
 }
 
+const POLLING_INTERVAL_MS = 5 * 1000; // 5 sec
+
+/**
+ * Hook for finding agents enrolled since component was rendered. Should be
+ * used by parent component to power rendering
+ * @param policyId
+ * @returns
+ */
+export const usePollingAgentCount = (policyId: string | undefined) => {
+  const initialIds: string[] = policyId ? [policyId] : [];
+
+  const [agentIds, setAgentIds] = useState(initialIds);
+
+  // Use useRef to guarantee we get the same date on each render
+  const mountedAt = useRef(Date.now());
+
+  useEffect(() => {
+    let timeout: NodeJS.Timeout;
+
+    const poll = () => {
+      timeout = setTimeout(async () => {
+        const secSinceMounted = Math.ceil((Date.now() - mountedAt.current) / 1000);
+        const request = await sendGetAgents({
+          kuery: `${AGENTS_PREFIX}.policy_id:"${policyId}" and ${AGENTS_PREFIX}.enrolled_at >= "now-${secSinceMounted}s"`,
+          showInactive: false,
+        });
+        const newAgentIds = request.data?.items.map((i) => i.id) ?? agentIds;
+
+        if (newAgentIds.some((id) => !agentIds.includes(id))) {
+          setAgentIds(newAgentIds);
+        }
+
+        poll();
+      }, POLLING_INTERVAL_MS);
+    };
+
+    poll();
+    return () => {
+      clearTimeout(timeout);
+    };
```

Comment:
There is a potential race condition here that could lead to an infinite polling, I think it could be solved by introducing an extra variable.

```suggestion
    let timeout: NodeJS.Timeout;
    let isAborted: boolean;

    const poll = () => {
      timeout = setTimeout(async () => {
        const secSinceMounted = Math.ceil((Date.now() - mountedAt.current) / 1000);
        const request = await sendGetAgents({
          kuery: `${AGENTS_PREFIX}.policy_id:"${policyId}" and ${AGENTS_PREFIX}.enrolled_at >= "now-${secSinceMounted}s"`,
          showInactive: false,
        });
        const newAgentIds = request.data?.items.map((i) => i.id) ?? agentIds;

        if (newAgentIds.some((id) => !agentIds.includes(id))) {
          setAgentIds(newAgentIds);
        }
        if (!isAborted) {
          poll();
        }
      }, POLLING_INTERVAL_MS);
    };

    poll();
    return () => {
      isAborted = true;
      clearTimeout(timeout);
    };
```